### ISSUE TITLE
"Shuffle Album" mode doesn't work when clementine was just started

### ISSUE THREAD

**Issue by robotanarchy** [clementine-player/Clementine#4402]
When you have the "Shuffle Album" mode selected, and Clementine just starts up, it will play the first track of the album and then randomly choose another album.

This also happens, when you switch the playlist. It plays the first track of the album fine, but then randomly chooses another one.

Thanks for that great music player :+1:

**Comment by enemymaker** [comment]
Unfortunately, it seems as this issue is not yet fully fixed.

I installed version 1.2.3-1346-gcc1b504 from clementine-dev (https://launchpad.net/~me-davidsansome/+archive/ubuntu/clementine-dev). And I experience the following behavior:

1) Set the shuffle mode to "Shuffle by album"
2) Start playing the playlist (let's assume we play "Song 3" from "Album 7")
3) Quit Clementine
4) Start Clementine
5) Resume the playlist (even though I have configured Clementine to always resume playback on start, I have to resume manually by clicking the play button)
6) After *Song 3* from *Album 7* finished, it plays always *Song 1* from *Album 7*, then *Song 2* from *Album 7* and so on. Interestingly, after *Song 6*, it skips *Song 7* and plays *Song 8* of *Album 7*.

Maybe this defect has been introduced by some other commit ...

Can someone confirm this behavior?

**Comment by GitHorst** [comment]
yes, can confirm (tested with Clementine 1.3.1 on both OS X 10.11.6 and Linux Mint 18.1).

to add to what was already described: once in shuffle album mode, it will play the song that you double-click but then continue from the 1st song of the album (instead of the song that would follow the double-clicked one), **ONLY** if nothing was playing before (i.e. you just started the program OR stopped the playback via the stop button). that is, once something is playing and you double click on any song to start it, it will continue with the expected behavior and not the bug described here.

thanks a lot for all your efforts with clementine!

**Comment by EnduringGuerila** [comment]
This was ages ago... But was it patched?

**Comment by GitHorst** [comment]
The original issue: yes, [my comment](https://github.com/clementine-player/Clementine/issues/4402#issuecomment-353991205): no
